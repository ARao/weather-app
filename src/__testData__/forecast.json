{
    "location": {},
    "current": {
        "condition": {}
    },
    "forecast": {
        "forecastday": [
            {
                "day": {
                    "condition": {}
                },
                "astro": {}
            },
            {
                "day": {
                    "condition": {}
                },
                "astro": {
                }
            }
        ]
    }
}