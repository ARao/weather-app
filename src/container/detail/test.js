import React from 'react';
import { shallow } from 'enzyme';
import ConnectedDetail from './index';
import configureStore from 'redux-mock-store'
import forecast from '../../__testData__/forecast.json'
import toJson from 'enzyme-to-json'

const fetchForecast = jest.fn()

describe('Detail', () => {

  const mockStore = configureStore();
  let wrapper, store;
  const initialState = {
    weather: {
      forecast
    }
  }

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<ConnectedDetail store={store} fetchForecast={fetchForecast} match={{ params: { id: 1 } }} />)
  })
  it('++++ render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('++++ store should match with value currentWeather', () => {
    expect(wrapper.props().currentWeather).toEqual(initialState.weather.current)
  });

  it('++++ store should match with value forecastWeather', () => {
    expect(wrapper.props().forecastWeather).toEqual(initialState.weather.forecast)
  });

  it('+++ snapshot match', () => {
    const snap = toJson(wrapper.dive());
    expect(snap).toMatchSnapshot();
  });
});
