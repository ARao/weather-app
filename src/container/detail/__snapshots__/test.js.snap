// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail +++ snapshot match 1`] = `
<div
  className="mt-4 "
>
  <Link
    replace={false}
    to="/"
  >
     
    <button
      className="btn btn-info"
    >
      Back
    </button>
  </Link>
  <div
    className="row"
  >
    <div
      className="mt-2 col-md-4 ml-auto"
    >
      <h5 />
      <p>
         
      </p>
      <img
        alt="icon"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-6 mx-auto"
    >
      <div
        className="pt-2"
      >
        <span
          className="mr-3"
        >
          Max Temp. 
        </span>
        <span>
           
          <sup>
            o
          </sup>
          C 
        </span>
      </div>
      <div
        className="pt-2"
      >
        <span
          className="mr-3"
        >
          Min Temp. 
        </span>
        <span>
           
          <sup>
            o
          </sup>
          C 
        </span>
      </div>
      <div
        className="pt-2"
      >
        <span
          className="mr-3"
        >
          Avg Temp. 
        </span>
        <span>
           
          <sup>
            o
          </sup>
          C 
        </span>
      </div>
      <div
        className="pt-2"
      >
        <span
          className="mr-3"
        >
          Max Wind 
        </span>
        <span>
           kmph 
        </span>
      </div>
      <div
        className="pt-2"
      >
        <span
          className="mr-3"
        >
          Avg Hummdity 
        </span>
        <span>
            
        </span>
      </div>
    </div>
  </div>
</div>
`;
